pkg/cmd/env/options.go

```diff
@@ -129,15 +131,27 @@ func (o *options) Run(f util.Factory) error {
 		}
 
 		t := o.CurrentTarget
-		if t.ShootName() == "" {
-			return target.ErrNoShootTargeted
-		}
 
 		client, err := manager.GardenClient(t.GardenName())
 		if err != nil {
 			return fmt.Errorf("failed to create garden cluster client: %w", err)
 		}
 
+		if t.ShootName() == "" {
+			if t.SeedName() != "" {
+				managedSeed, err := client.GetManagedSeed(f.Context(), t.SeedName())
+				if err != nil && !apierrors.IsNotFound(err) {
+					return err
+				}
+
+				if managedSeed != nil {
+					o.CurrentTarget = o.CurrentTarget.WithProjectName("garden").WithShootName(managedSeed.Name)
```

Comment:
The currentTarget should not be changed. Or we should rename the property to Target.

Reply:
The managedSeed has a namespace which we use in case of getting the clientConfig. Here we ignore the namespace an hardcode the the project name to be `garden`

Reply:
Yes, I agree. I also think that we should rename it. If we leave the code at that place, we should definitively do it.

Reply:
>The managedSeed has a namespace which we use in case of getting the clientConfig. Here we ignore the namespace an hardcode the the project name to be garden

The `Seed` resource is not namespaces, however the `ManagedSeed` resource is. Don't ask me why because it is only possible to create `ManagedSeed` resources in the garden namespaces. Of course, shoots that are used as seeds are (currently) always in the garden namespace.
See also here: https://github.com/gardener/gardener/blob/master/docs/usage/managed_seed.md
>Unlike the Seed resource, the ManagedSeed resource is namespaced. Currently, managed seeds are restricted to the garden namespace.

That's why I hard-coded it. Do you have a cleaner solution?

Reply:
I renamed the current target